homeassistant/components/automation/__init__.py

```diff
@@ -569,11 +571,16 @@ async def _async_attach_triggers(
         def log_cb(level, msg, **kwargs):
             self._logger.log(level, "%s %s", msg, self.name, **kwargs)
 
-        variables = None
+        self.async_write_ha_state()
+        state = self.hass.states.get(self.entity_id)
+        assert state is not None
+        variables = {"this": state.as_dict()}
```

Comment:
```suggestion
        state = self.hass.states.get(self.entity_id)
        this = None
        if state is not None:
            this = state.as_dict()
        variables = {"this": this}
```

Reply:
This is where I found state to be none in some unit tests  ... I and used my previous solution, similar to yours. But Eric suggested I add the  self.async_write_ha_state() and again use  the assert to get rid of the lint warning.